src/dysh/spectra/spectrum.py

```diff
@@ -931,15 +934,70 @@ def _div_meta(self, operand, operand2=None, **kwargs):
 
     def __getitem__(self, item):
 
-        tmp_spec = super().__getitem__(item)
-        trimmed = tmp_spec._copy(
-            spectral_axis=tmp_spec.spectral_axis,
-            wcs=None,
-            meta=tmp_spec.meta,
-            observer=self.observer,
-            target=self.target,
-        )
-        return trimmed
+        def q2idx(q, wcs, spectral_axis, coo, sto):
+            """Quantity to index."""
+            if "velocity" in u.get_physical_type(q):
+                # Convert from velocity to channel.
+                idx = vel2idx(q, wcs, spectral_axis, coo, sto)
+            elif "frequency" in u.get_physical_type(q):
```

Comment:
PhysicalType('length')  is also valid for wavelength units

Reply:
s5 = ta[211.8*u.mm:212.8*u.mm]
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
Cell In[33], line 1
----> 1 s5 = ta[211.8*u.mm:212.8*u.mm]

File ~/src/dysh/src/dysh/spectra/spectrum.py:986, in Spectrum.__getitem__(self, item)
    984 # Start.
    985 if isinstance(start, u.Quantity):
--> 986     start_idx = q2idx(start, wcs, spectral_axis, coo, sto)
    987 else:
    988     start_idx = start

File ~/src/dysh/src/dysh/spectra/spectrum.py:952, in Spectrum.__getitem__.<locals>.q2idx(q, wcs, spectral_axis, coo, sto)
    950     idxs = wcs.world_to_pixel(coo, q, sto)
    951     idx = int(np.round(idxs[0]))
--> 952 return idx

UnboundLocalError: local variable 'idx' referenced before assignment

Reply:
Good catch! Give it a try now.